src/Native/System.Native/pal_networking.cpp

```diff
@@ -379,7 +380,119 @@ static void ConvertHostEntPlatformToPal(HostEntry& hostEntry, hostent& entry)
     }
 }
 
-#if HAVE_GETHOSTBYNAME_R
+#if !HAVE_THREAD_SAFE_GETHOSTBYNAME_AND_GETHOSTBYADDR && (!HAVE_GETHOSTBYNAME_R && !HAVE_GETHOSTBYADDR_R)
```

Comment:
In that case, could we have a single define we use for both? Or do they need to be separate for some reason?